desktop/cmp/persistenceManager/PersistenceManagerModel.ts

```diff
@@ -0,0 +1,345 @@
+ * PersistenceManager provides re-usable loading, selection, and user management of named configs, which are modelled
+ * and persisted on the server as databased domain objects extending the `PersistedObject` trait.
+ *
+ * These generic configs are intended for specific use cases such as saved Grid views, Dashboards, and data import
+ * mapping configs. This model loads all available views from a configured endpoint and exposes a `provider` property
+ * that can be passed to any HoistModel that takes a `persistWith` config.
+ *
+ * Objects managed by this system can be private or shared based on a user's company and/or app roles. The `acl` field
+ * on each persisted object determines if and how it is shared. Users with the configured `adminRole` can share saved
+ * objects to all app users and save changes to shared objects directly. Users with the configured `editorRole` can
+ * share objects to their own company only and modify shared objects shared only within their company. All users can
+ * save/update/delete their own private objects.
+ */
+export interface PersistenceManagerProps {
+    // /** Endpoint for rest controller. */
+    // url: string;
+
+    /** Key used in JsonBlob */
+    type: string;
+    /** User-facing name/label for an object managed by this model. */
+    noun: string;
+    /** Whether user can publish or edit globally shared objects. */
+    canManageGlobal: Thunkable<boolean>;
+    /** Async callback triggered when view changes. Should be used to recreate the affected models. */
+    onChangeAsync: (value: PlainObject) => void;
+    /** Used to persist this model's selected ID. */
+    persistWith: PersistOptions;
+    /** True (default) to render a save button alongside the primary menu button when dirty. */
+    enableTopLevelSaveButton?: boolean;
+    /** Fn to produce a new, empty object - can be async. */
+    newObjectFnAsync?: () => PlainObject;
+}
+
+export class PersistenceManagerModel extends HoistModel {
+    //------------------------
+    // Persistence Provider
+    // Pass this to models that implement `persistWith` to include their state in the view.
+    //------------------------
+    provider: PersistOptions = {
+        getData: () => cloneDeep(this.pendingValue ?? this.value ?? {}),
+        setData: value => this.mergePendingValue(value)
+    };
+
+    @observable.ref @managed manageDialogModel: ManageDialogModel;
+
+    @observable.ref @managed saveDialogModel: SaveDialogModel;
+
+    enableTopLevelSaveButton: boolean = true;
+
+    newObjectFn: () => PlainObject;
+
+    /** Current state of the active object, can include not-yet-persisted changes. */
+    @observable.ref pendingValue: PlainObject = null;
+
+    @observable.ref views: PlainObject[] = [];
+
+    @bindable selectedId: StoreRecordId;
+
+    /** Reference Key used to query JsonBlobs */
+    type: string;
+
+    /** Configured word to label an object persisted by this model. */
+    noun: string;
+
+    get pluralNoun(): string {
```

Comment:
Not sure these methods on the model are adding much value. Could add `@computeOnce` decorators if you want to cache these. Otherwise, I think I would just call the lodash methods directly?